src/ucs/datastruct/ptr_array.c

```diff
@@ -54,16 +49,39 @@ ucs_ptr_array_freelist_set_next(ucs_ptr_array_elem_t *elem, unsigned next)
                     (((ucs_ptr_array_elem_t)next) << UCS_PTR_ARRAY_NEXT_SHIFT);
 }
 
+static inline unsigned
+ucs_ptr_array_freelist_get_next_params(ucs_ptr_array_elem_t elem,
```

Comment:
this function is not needed:
- its interface is weird, it returns 2 params, one as a retvalue, another by pointer
- we have two other funcs for getting size and next
- seems like you do not have a use case where you use both values returned by this func

Reply:
The reason why we need this function is that calling get_next() and then get_size_ahead() would yield a slower code - To optimize the access to the element's memory, I decided to unite the parsing of the element in a single function, if required.
See an example for usage in line 216.

Reply:
you do not need this func on line 216, because there you need just `size_free_ahead`, but not `next`

Reply:
OK, I see what you mean - I removed the get_next_params() function and replaced it with get_size_free_ahead().